module/os/freebsd/zfs/zfs_vnops_os.c

```diff
@@ -5626,7 +5626,7 @@ zfs_setextattr_sa(struct vop_setextattr_args *ap, const char *attrname)
 		error = nvlist_add_byte_array(nvl, attrname, buf, entry_size);
 	kmem_free(buf, entry_size);
 	if (error == 0)
-		error = zfs_sa_set_xattr(zp);
+		error = zfs_sa_set_xattr(zp, attrname, buf, entry_size);
```

Comment:
`buf` was just freed, let's move that to after this last use.

Reply:
Thanks. Done.